wallbe/devicepluginwallbe.cpp

```
@@ -0,0 +1,197 @@
+    update(device);
+
+    if (!m_pluginTimer) {
+        m_pluginTimer = hardwareManager()->pluginTimerManager()->registerTimer(120);
+        connect(m_pluginTimer, &PluginTimer::timeout, this, [this] {
+            foreach(Device *device, m_connections.keys()) {
+                update(device);
+            }
+        });
+    }
+}
+
+void DevicePluginWallbe::executeAction(DeviceActionInfo *info)
+{
+    Device *device = info->device();
+    Action action = info->action();
+
+    WallBe *wallbe = m_connections.value(device);
+    if (!wallbe) {
+        info->finish(Device::DeviceErrorHardwareFailure);
+        return;
+    }
+
+    if (device->deviceClassId() == wallbeEcoDeviceClassId){
+
+        // check if this is the "set power" action
+        if (action.actionTypeId() == wallbeEcoPowerActionTypeId) {
+
+            // get the param value of the charging action
+            bool charging =  action.param(wallbeEcoPowerActionPowerParamTypeId).value().toBool();
+            qCDebug(dcWallbe) << "start Charging button" << device->name() << "set power to" << charging;
+            wallbe->setChargingStatus(charging);
+            // Set the "power" state
+            device->setStateValue(wallbeEcoPowerStateTypeId, charging);
+            info->finish(Device::DeviceErrorNoError);
+            return;
+
+        } else if(action.actionTypeId() == wallbeEcoChargeCurrentActionTypeId){
+
+            uint16_t current = action.param(wallbeEcoChargeCurrentEventChargeCurrentParamTypeId).value().toInt();
+            qCDebug(dcWallbe) << "Charging power set to" << current;
+            wallbe->setChargingCurrent(current);
+            device->setStateValue(wallbeEcoChargeCurrentStateTypeId, current);
+            info->finish(Device::DeviceErrorNoError);
+            return;
+        }
+        info->finish(Device::DeviceErrorActionTypeNotFound);
+        return;
+    }
+    info->finish(Device::DeviceErrorDeviceClassNotFound);
```

I'm still undecided whether this is a good idea or not... In theory it should *never* be called anyways, And if it does, it sure is an implementation error in the plugin (well, or a bug in the core, which, let's assume, doesn't happen).

So if, by any chance you run into this, you really want to fix your code (or tell the core devs ASAP if the core calls this with an invalid deviceClassId to get it fixed in there).

So actually I think instead of just silently making this executeAction call fail, it would make more sense to use an assertion like:
```
Q_ASSERT_X(false, "executeAction", QString("Unhandled action: %1).arg(action.name().toUtf8());
```

In this case it would crash your development build. In a release build it would make the action time out in the core eventually.